Sources/Remote Logging/Crash Logging/CrashLogging.swift

```diff
@@ -187,8 +187,12 @@ public extension CrashLogging {
                 scope.setLevel(level)
             }
         }
-        SentrySDK.flush(timeout: CrashLogging.eventsFlushingTimeout)
-        callback()
+
+        let queue = DispatchQueue(label: "com.automattic.tracks.flushSentry", qos: .background)
+        queue.async {
+            SentrySDK.flush(timeout: CrashLogging.eventsFlushingTimeout)
+            callback()
+        }
```

Comment:
@crazytonyli what do you think of this as an implementation of your suggestion to make the call in a background queue?

Reply:
In regards to your suggestion to [add a `wait: Bool`](https://github.com/Automattic/Automattic-Tracks-iOS/pull/232#discussion_r999047433) parameter, are you thinking of something along the lines [of WordPress iOS](https://github.com/wordpress-mobile/WordPress-iOS/blob/feb1cc189cf70c050f8bc5aa74630d4b9dff34a2/WordPress/Classes/Utility/ContainerContextFactory.swift#L23-L35) where we use a `Bool` parameter to decide whether to call `context.performAndWait(_:)` or `context.perform(_:)`?

In this, I'm thinking if `wait == true` we wouldn't run the code in `queue.async`.

Reply:
Yes, so that we don't need to use semaphore like the original implementation 😄

Reply:
😕 The new diff makes this conversation odd because the code it referred to has moved. Anyways... @crazytonyli I had an initial go at adding a `wait` parameter. 

I did this in the form of a `private` func for the other methods to call.

```swift
private func logErrorsImmediately(
        _ errors: [Error],
        userInfo: [String: Any]? = nil,
        level: SentryLevel = .error,
        andWait wait: Bool,
        callback: @escaping () -> Void
    ) {
```

I didn't modify the original method signatures to avoid a breaking change, but also because I wasn't sure how it would have felt to have a `andWait:,` together with a `callback:`. If `wait == true`, then there's no need for a callback because the code that would run in there essentially runs once the method has finished anyway.